### PR TITLE
optimization for loop

### PR DESCRIPTION
**Pull request by shellhub**

### CODE REVIEW

File: okhttp/src/main/java/okhttp3/Headers.java [hunk 1]
```diff
@@ -365,7 +364,7 @@ Builder addLenient(String name, String value) {
     }
 
     public Builder removeAll(String name) {
-      for (int i = 0; i < namesAndValues.size(); i += 2) {
+      for (int i = 0, size = namesAndValues.size(); i < size; i += 2) {
```
[the remark this answers is not in the document]
Comment (shellhub) [reply okhttp/src/main/java/okhttp3/Headers.java]: Yes.

[the remark this answers is not in the document]
Comment (shellhub) [reply okhttp/src/main/java/okhttp3/Headers.java]: please review again @swankjesse

[the remark this answers is not in the document]
Comment (shellhub) [reply okhttp/src/main/java/okhttp3/Headers.java]: yes. example code
``` java
Headers.Builder headers = new Headers.Builder();
headers.add("name", "value1");
headers.add("name", "value2");
headers.add("name", "value3");
headers.add("name", "value4");
headers.add("name", "value5");

headers.removeAll("name");
System.out.println(headers.namesAndValues);
```

File: okhttp/src/main/java/okhttp3/ConnectionPool.java [hunk 2]
```diff
@@ -255,7 +255,7 @@ long cleanup(long now) {
    */
   private int pruneAndGetAllocationCount(RealConnection connection, long now) {
     List<Reference<StreamAllocation>> references = connection.allocations;
-    for (int i = 0; i < references.size(); ) {
+    for (int i = 0, size = references.size(); i < size; ) {
```
Comment (swankjesse) [inline okhttp/src/main/java/okhttp3/ConnectionPool.java]: This is incorrect; references’ size changes on line 273.

File: okhttp/src/main/java/okhttp3/Headers.java [hunk 3]
```diff
@@ -365,11 +364,12 @@ Builder addLenient(String name, String value) {
...
         if (name.equalsIgnoreCase(namesAndValues.get(i))) {
           namesAndValues.remove(i); // name
           namesAndValues.remove(i); // value
           i -= 2;
+          size -= 2;
```
Comment (swankjesse) [inline okhttp/src/main/java/okhttp3/Headers.java]: My preference is to just do a normal for loop whenever the size changes in the loop body.

**Pull request merged**